src/table/body-cell/index.tsx

```diff
@@ -56,6 +56,28 @@ function TableCellEditable<ItemType>({
   const [hasFocus, setHasFocus] = useState(false);
   const showIcon = hasHover || hasFocus;
 
+  const prevSuccessfulEdit = useRef(successfulEdit);
+  const prevHasFocus = useRef(hasFocus);
+  const [showSuccessIcon, setShowSuccessIcon] = useState(false);
+
+  useEffect(() => {
+    // Early return if there hasn't been any successfulEdit state changes.
+    if (!successfulEdit && !prevSuccessfulEdit.current) {
+      return;
```

Comment:
If we exit early because of `successfulEdit`, doesn't that mean that we sometimes dont update `prevSuccessfulEdit`? I think this will be fixed automatically we switch to our `usePrevious` hook.